src/ast/value.rs

```diff
@@ -61,6 +63,37 @@ pub enum Value {
     Null,
 }
 
+impl Value {
+    pub fn to_negative(self) -> Self {
+        match self {
+            #[cfg(not(feature = "bigdecimal"))]
+            Value::Number(x, v) => Value::Number(format!("-{}", x), v),
```

Comment:
@PsiACE  Do you why `format` is not allowed here but it can be allowed in other places.

Reply:
`format!` allocates a `String`, and a `no_std` environment doesn't have an allocator.

https://stackoverflow.com/questions/50200268/how-can-i-use-the-format-macro-in-a-no-std-environment

Reply:
but there are `format!` in other places of this repo.

Reply:
i will fix it later

Reply:
Done, use `alloc::format`